app/controllers/treatments/treatments_controller.rb

```diff
@@ -0,0 +1,40 @@
+class Treatments::TreatmentsController < ApplicationController
+
+  def create
+    start_treatment
+    respond_to do |format|
+      if user.save
+        format.html { redirect_to users_path, notice: "#{human_controller_name} created for #{user.decorate.full_name}"}
+      else
+        flash[:alert] = user.errors.full_messages
+        format.html { render action: "new", status: :unprocessable_entity}
+      end
+    end
+  end
+
+  # @param ARR [User]
+  # @param Array params
+  # @return ARR [Trim]
+  def start_treatment
+    UserTreatment.new.call(user: user, type: trim_type)
+  end
+
+  protected
+
+  def human_controller_name
+    controller_name.humanize.singularize
+  end
+
+  def user
+    @user ||= User.find(params[:user_id])
+  end
+
+  def treatment_type_params
```

Comment:
Hmmm.  So this is a bad practice which we have outlawed in software development here at PH.  This is what we refer to as "Abstract Template Classing"..
https://dockyard.com/blog/ruby/2013/07/10/design-patterns-template-pattern

It makes code flow extremely difficult to figure out.  You are making a contract with the subclasses who use you that they must implement a private method without much understanding as to why.